opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/requisition/RequisitionInterface.java

```diff
@@ -498,4 +517,35 @@ public int compareTo(final RequisitionInterface other) {
             .append(m_description, other.m_description)
             .toComparison();
     }
+
+    @SuppressWarnings("unused")
+    public void beforeMarshal(Marshaller u) {
+        if (m_ipAddressStr == null && m_ipAddress != null) {
+            m_ipAddressStr = InetAddressUtils.str(m_ipAddress);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public void afterUnmarshal(Unmarshaller u, Object parent) throws IPValidationException {
+        validateHost();
+    }
+
+    /**
+     *  Post-unmarshall, this is called to construct the InetAddress
+     *  from the parsed ip-addr String.
+     *
+     * @throws IPValidationException If the parsed ip-addr attribute
+     *                               does not resolve to a valid address
+     */
+    protected void validateHost() throws IPValidationException {
+        try {
+            m_ipAddress = new IPAddress(m_ipAddressStr).toInetAddress();
+        }
+        catch (IllegalArgumentException iae) {
+            final Event e = EventUtils.createInterfaceRejectedEvent("Provisiond", m_ipAddressStr);
+            LOG.warn("Invalid IP address found in a requisition interface: {}", m_ipAddressStr);
+            StaticEventForwarder.getStaticEventForwarder().sendNow(e);
```

Comment:
I don't like the way event forwarder is wired here.  We shouldn't wire in any dependencies in model classes.  Clients that load `RequisitionInterface` should handle validation and try to send event.

Reply:
I tried to catch the validation issues in other locations, but there are code paths to ingest Requisition objects from the filesystem that I couldn't get to reliably validate the object before using it. Specifically, [this test](https://github.com/OpenNMS/opennms/blob/73e9ad89621078003d629444d7c90a46600f2378/opennms-provision/opennms-provision-persistence/src/test/java/org/opennms/netmgt/provision/persist/FilesystemForeignSourceRepositoryTest.java#L112) was continually failing when comparing the local Requistion to the one gotten from the repository.

The desired behavior from [NMS-13806](https://issues.opennms.org/browse/NMS-13806) is to just ignore any invalid interfaces and remove them from the model before ingesting. Throwing an error back to a client trying to ingest a large amount of nodes would violate this, as the parsing would be stopped. I tried to do all of the validation in the validate() methods that all of the Requisition model objects have, but the test I reference above was failing, and I couldn't figure out why the Requisition retrieved from the repository wasn't getting properly validated. 

The only reliable way to catch the issue that i found was to have the [ValidationEventHandler](https://github.com/OpenNMS/opennms/pull/4195/files#diff-6929f4c77301a59258c4200dd4197c145eebe41c4f5e61c15d3cddd923664afcL98) catch it and attempt to continue parsing, so the issue is only known within the model object.

I don't necessarily like the wiring either, but this is the only way I could think of to get an event out with this information in it. Another option is to drop the event entirely and just log the issue and move on.

Reply:
I would opt to remove sending event from model objects. Catch specific validation exception and log it with `WARN` and move on.